### ISSUE TITLE
How to use drawImage?

### ISSUE THREAD

**Issue by noughtmare** [arrayfire/arrayfire-haskell#13]
Your documentation contains an example:

```Haskell
>>> drawImage window ('constant' \@'Int' 1) ('Cell' 10 10 "test" 'ColorMapSpectrum')
```

I adapted it to:
```Haskell
main = do
  w <- createWindow 512 512 "Test"
  drawImage w (constant @Int [512,512] 1) (Cell 10 10 "test" ColorMapSpectrum)
```

That gives the error:
```
AFException {afExceptionType = TypeError, afExceptionCode = 204, afExceptionMsg = "Function does not support this data type"}
```

**Comment by dmjio** [comment]
@noughtmare I'd try using the `loadImage` function, if that still gives you problems I can look into it tonight. The example I have might be outdated.

**Comment by dmjio** [comment]
@noughtmare I think the type needs to be `CBool` (`b8`). Can confirm soon.

**Comment by noughtmare** [comment]
@dmjio If I use `CBool` then I get the error:
```
free(): invalid size
Aborted
```

**Comment by dmjio** [comment]
@noughtmare there's been some fixes (related to double free'ing of pointers) that are now in HEAD, am working on the Graphics module now with success. `drawSurface` in particular.

As I inspect more with what the C++ is doing before it sends it to the C, we'll have better working visualization code, and examples !

```haskell
main :: IO ()                                                                                                                                                                                                   
main = do                                                                                                                                                                                                       
  window <- createWindow 800 600 "hey"                                                                                                                                                                          
  let x = iota [60,1] [1,60] / 29                                                                                                                                                                               
      y = iota [1,60] [60,1] / 29                                                                                                                                                                               
  go window (0 :: Array Float) x y                                                                                                                                                                              
    where                                                                                                                                                                                                       
      go window t x y = do                                                                                                                                                                                      
        let z = 10*x*(-abs(y)) * cos(x*x*(y+t))+sin(y*(x+t))-1.5                                                                                                                                                
        drawSurface window x y z (Cell (-1) (-1) "there" ColorMapDefault)                                                                                                                                       
        closed <- isWindowClosed window                                                                                                                                                                         
        unless closed $ go window (t + 0.07) x y  
```

![image](https://user-images.githubusercontent.com/875324/68562123-4fa18b00-0416-11ea-8e10-6e5d9748bdf4.png)

**Comment by dmjio** [comment]
@noughtmare, for `drawImage`, try something like this:

```haskell
main :: IO ()                                                                                                                                                                                                   
main = render =<< createWindow 800 600 "hey"                                                                                                                                                                          
    where                                                                                                                                                                                                       
      render window = do                                                                                                                                                                                            
        let x = iota [1,60] [60,1] / (30 :: Array Float)                                                                                                                                                        
            cell = Cell (-1) (-1) "" ColorMapDefault                                                                                                                                                        
        drawImage window x cell                                                                                                                                                                                 
        closed <- isWindowClosed window                                                                                                                                                                         
        unless closed (render window )
```
![image](https://user-images.githubusercontent.com/875324/68563080-7497fd00-041a-11ea-8c1e-049549a2dca2.png)

I'll update the documentation to reflect this.
